compiler/damlc/daml-stdlib-src/DA/Internal/Interface.daml

```diff
@@ -0,0 +1,49 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | MOVE Prelude interface functionality
+module DA.Internal.Interface (
+  HasInterfaceTypeRep(..),
+  HasToInterface(..),
+  HasFromInterface(..),
+  Implements,
+  toInterface,
+  toInterfaceContractId,
+  fromInterfaceContractId,
+) where
+
+import DA.Internal.Prelude
+import DA.Internal.Template.Functions
+import DA.Internal.LF
+
+class HasInterfaceTypeRep i where
+  interfaceTypeRep : i -> TypeRep
+
+class HasToInterface t i where
+  _toInterface : t -> i
+
+toInterface : forall i t. HasToInterface t i => t -> i
```

Comment:
same question as on the GHC PR. Why do we need a separate top-level method here?

Reply:
we need it to flip the type arguments. We could flip the arguments to the `HasToInterface` class, but @sofiafaro-da and I found it a bit confusing to have some classes take `t i` and others `i t`.

Reply:
oh I completely missed the fact that we have different argument order. maybe worth adding a comment

Reply:
will do!

Reply:
done now :)